generators/client/templates/angular/src/main/webapp/app/blocks/interceptor/_auth-expired.interceptor.ts

```diff
@@ -36,81 +36,82 @@ import { LoginModalService } from '../../shared/login/login-modal.service';
 import { StateStorageService } from '../../shared/auth/state-storage.service';
 <%_ } _%>
 
-export class AuthExpiredInterceptor extends JhiHttpInterceptor {
+export class AuthExpiredInterceptor implements HttpInterceptor {
 
 <%_ if (authenticationType === 'jwt' || authenticationType === 'uaa') { _%>
     constructor(private injector: Injector) {
-        super();
     }
 <%_ } else if (authenticationType === 'session') { _%>
     constructor(
         private injector: Injector,
         private stateStorageService: StateStorageService,
         private loginServiceModal: LoginModalService) {
-        super();
     }
 <%_ } else if (authenticationType === 'oauth2') { _%>
     constructor(private injector: Injector,
         private stateStorageService: StateStorageService) {
-        super();
     }
 <%_ } _%>
 
-    requestIntercept(options?: RequestOptionsArgs): RequestOptionsArgs {
-        return options;
-    }
 <%_ if (authenticationType === 'jwt' || authenticationType === 'uaa') { _%>
-
-    responseIntercept(observable: Observable<Response>): Observable<Response> {
-        return <Observable<Response>> observable.catch((error, source) => {
-            if (error.status === 401) {
-<%_ if (authenticationType === 'jwt') { _%>
-                const loginService: LoginService = this.injector.get(LoginService);
-                loginService.logout();
-<%_ } _%>
-<%_ if (authenticationType === 'uaa') { _%>
-                const principal = this.injector.get(Principal);
-
-                if (principal.isAuthenticated()) {
-                    principal.authenticate(null);
-                    const loginModalService: LoginModalService = this.injector.get(LoginModalService);
-                    loginModalService.open();
-                } else {
+    intercept(request: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
+        return next.handle(request).do((event: HttpEvent<any>) => {
+            if (event instanceof HttpResponse) {
+                // success
+            }
+        }, (err: any) => {
+            if (err instanceof HttpErrorResponse) {
+                if (err.status === 401) {
+<% if (authenticationType === 'jwt') { %>
                     const loginService: LoginService = this.injector.get(LoginService);
```

Comment:
Is there any possibility to remove the injector completely? This looks like hacky, can't we inject these services straight

Reply:
will try that tomorrow it should at least work for login service not sure what happens with Principal

Reply:
Ok tried to remove it and it doesnt work as it woudl create a cyclic dependenncy becuase login service uses HttpClient that uses the interceptor

Reply:
yes, Injector is used to avoid cyclic dependency hell in angular